### PR TITLE
Enumerables: Release/0.1.0

### PR DESCRIPTION
**Pull request by Kasuhira**
Hello, reviewer, how are you?
I'm releasing the first version of the Enumerables algoritms, the focus is to make it usable and learn about git flow, I'm having dificulties on finding more bugs so I ask for guidance.
When ready, I shall merge, happy review !

### CODE REVIEW

File: enumerable_methods.rb [hunk 1]
```diff
@@ -162,6 +164,8 @@ def multiply_els(arr)
   arr.my_inject(1) { |product, result| product * result }
 end
 
+puts longest
```
Comment (whiz25) [inline enumerable_methods.rb]: Where is this `longest` coming from?
longest

File: enumerable_methods.rb [hunk 2]
```diff
@@ -90,14 +90,16 @@ def my_none?(exp = nil)
...
     if block_given?
       my_each { |x| counter += 1 if yield(x) }
-      counter
     else
-      my_each { counter += 1 }
+      my_each do |x|
+        counter += 1 if num == x
+      end
     end
+    counter
```
Comment (whiz25) [inline enumerable_methods.rb]: - [ ] should be identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns the number of items in enum through enumeration
- [ ] counts the number of elements yielding a true value if a block is given

File: enumerable_methods.rb [hunk 3]
```diff
@@ -115,10 +117,10 @@ def my_map(proc = nil)
   end
 
   # rubocop:disable Metrics/MethodLength
-  def my_inject(ind = nil, symb = nil)
+  def my_inject(ind = to_a[0], symb = nil)
```
Comment (whiz25) [inline enumerable_methods.rb]: - [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] combines all elements of enum by applying a binary operation, specified by a block
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method

File: enumerable_methods.rb (hunk 2 above)
Comment (Kasuhira) [reply enumerable_methods.rb]: Solved !

File: enumerable_methods.rb [hunk 4]
```diff
@@ -90,14 +90,16 @@ def my_none?(exp = nil)
...
       my_each { |x| counter += 1 if yield(x) }
-      counter
     else
-      my_each { counter += 1 }
+      my_each do |x|
+        counter += 1 if num == x
+      end
     end
+    counter
   end
```
[the remark this answers is not in the document]
Comment (Kasuhira) [reply enumerable_methods.rb]: Oh, nice! Fixed !

File: enumerable_methods.rb [hunk 5]
```diff
@@ -115,17 +117,17 @@ def my_map(proc = nil)
...
       if ind.is_a? Symbol
         case ind
         when :+
-          counter = 0
-          my_each { |x| counter += x }
+          counter = self[0]
+          self[1..-1].my_each { |x| counter += x }
         when :-
           counter = self[0]
           self[1..-1].my_each { |x| counter -= x }
```
[the remark this answers is not in the document]
Comment (Kasuhira) [reply enumerable_methods.rb]: Fixed!

**Comment by Anna-Myzukina** [comment]
# Status [ APPROVED] image

I tested your methods and all cases passed.

### You can merge it. And don't forget to fill the project completion form.





*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*

**Pull request merged**